junit4/src/main/java/org/jboss/weld/junit4/WeldInitiator.java

```diff
@@ -213,6 +219,35 @@ private WeldInitiator(Weld weld, List<Object> instancesToInject, Set<Class<? ext
         super(weld, instancesToInject, scopesToActivate, beans, resources, ejbFactory, persistenceUnitFactory, persistenceContextFactory);
     }
 
+    /**
+     * Returns a {@link MethodRule} that can be used as a {@link Rule @Rule} to inject into the test class instance.
+     * When using this weld initiator as {@code @Rule}, the same can be achieved by calling
+     * {@link AbstractBuilder#inject(Object) inject(this)} on the builder. But when using this weld initiator as
+     * {@link ClassRule @ClassRule}, there is no {@code this} reference available. In that case this method can be used
+     * as {@code @Rule} to do the test class injection part.
+     *
+     * <p><b>Example:</b>
+     * <pre>{@code
+     * }&#64;{@code ClassRule
+     * public static WeldInitiator weld = WeldInitiator.from(Foo.class).build();
+     *
+     * }&#64;{@code Rule
+     * public MethodRule testClassInjector = weld.getTestClassInjector();
+     * }</pre>
+     *
+     * @return the test class injector rule
+     */
+    public MethodRule getTestClassInjector() {
```

Comment:
I was thinking we could come up with some more spot on name for this method. People will use it exclusively to achieve injection under this scenario, so something like `performTestClassInjection()`? WDYT?

Reply:
Well, I thought the name I chose **is** spot on actually. :-D
The method does not perform any injection. It just gives you an instance, that if called as rule (or manually) will do the injection.
So imho, the name is perfectly appropriate, with `perform...` users could assume the injection is done already and don't get, that they have to use the returned value, for example as rule like intended.

After all it is your project, so I'm gong to change it if you want, but imho this would not improve it, but the contrary.

Reply:
> After all it is your project

I tend not to think of myself as a dictator, so opposing opinions are very much welcome ;-)

I am fine keeping it as-is if you feel like the name is clear enough for users needing this functionality.

Reply:
`getTestClassInjector()` is fine. I'd even vote for ` getTestClassInjectorRule()`...